File: backend/src/routes/aiRoutes.js
// routes/aiRoutes.js
import express from "express";
import { askAI, askAIWithOllama } from "../controllers/aiController.js";

const router = express.Router();

router.post("/ask-ai", askAI);
router.post("/ask-ollama", askAIWithOllama);

export default router;

File: backend/app/routes/ai_routes.py
from flask import Blueprint, request, jsonify
from controllers.ai_controller import ask_ai_with_ollama
from middlewares.authenticate import  require_api_key

ai_bp = Blueprint('ai_bp', __name__)


@ai_bp.route('/ask-ollama', methods=['POST'])
@require_api_key
def handle_ask_ollama():
    return ask_ai_with_ollama(request)
